test/C/AsyncCallback2.lf

```diff
@@ -32,7 +31,10 @@ main reactor AsyncCallback2 {
             struct timespec sleep_time = {(time_t) 0, (long)10000000};
             struct timespec remaining_time;
             nanosleep(&sleep_time, &remaining_time);
-            callback((trigger_t*)a);
+            // a is actually a pointer to the action struct on the self struct.
+            // But the first item in that struct is a pointer to the trigger_t
+            // object. The following song and dance gets that pointer.
+            callback(*(trigger_t**)a);
```

Comment:
Actually, this test has illegal code: it is asynchronously invoking a logical action. The fact that the test has to this tricky thing is further evidence that we must enforce this policy. The runtime should exit with an error when schedule is called in this test.

Reply:
I agree that asynchronously scheduling a logical action should be treated as an error. Would the expression `*(trigger_t**)a` be any different for a physical action though?

Reply:
No, to schedule physical actions, you don't call __schedule() (an internal function that takes a trigger_t*), but rather call one of the other schedule() functions, which take a void* pointer to the action.  The reason this test calls __schedule() is because it needs to hold the mutex lock for two successive calls of __schedule(). Otherwise, the test intermittently fails.  This is further indication that what this test is doing is not kosher.

Reply:
Right... I think the point here was to ensure that of two events scheduled at the same logical time only the last-scheduled one would survive. There is no need (or valid use case) for doing this asynchronously, however. We should just have two subsequent calls to schedule within a reaction to test this. I agree: this test needs to be revised.